### ISSUE TITLE
Add GH actions to automatically run pre-commit 

### ISSUE THREAD

**Issue by kaitejohnson** [CDCgov/ww-inference-model#139]
##Goal
Currently have to pull locally and run `pre-commit --all-files` to fix this, which can be slow and annoying especially in final stages of review with a number of minor tweaks. @zsusswein has set this up to be done [automatically via github actions](https://github.com/CDCgov/cfa-gam-rt/commit/ff136bfc11e7347e2e82244524a17ecf5d314b79#diff-3c0d80ea54e617ba55ba031dda6fc983852272f93775375ce402a66a03560548R37)

**Comment by zsusswein** [comment]
See the workflow file with the change here: https://github.com/CDCgov/cfa-epinow2-pipeline/pull/18

**Comment by dylanhmorris** [comment]
Adjusted the title and text a bit to be more specific. @kaitejohnson can you confirm that this is what you're looking to do.

**Comment by kaitejohnson** [comment]
Yeah, I thought it was a quick fix though and I am not sure it is so leaving as a draft. Feels low priority.

**Comment by zsusswein** [comment]
I did two things:

1. Pre-commit hooks run in CI and auto-commit & -push changes on failure
    - This doesn't include roxygen because that particular hook requires manually keeping a second, duplicate package manifest in the `.pre-commit.config.yaml` and I refuse to do that. It also breaks if the package has any non-R (system) dependencies.....like Stan.
1. I set up roxygen2 render and code styling to trigger via a button push (`workflow_dispatch`) and/or a comment with a slash command in the PR. I don't want to do all the re-rendering locally and we have frequent suggested changes to documentation.

As Sam is saying, the auto-commit might be a mistake if the team isn't comfortable with rebasing pulls. I'm currently the only one consistently committing to this project, but will re-evaluate down the road if it becomes an issue.
